### ISSUE TITLE
Implement Basic Neural Network Layers

### ISSUE THREAD

**Issue by Shashwatpandey4** [Shashwatpandey4/Cortex#4]
Core layers needed for initial networks:
- [ ] Linear (Dense) layers
- [ ] Conv2d
- [ ] MaxPool2d, AvgPool2d
- [ ] BatchNorm2d
- [ ] Dropout
